modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableColumnHeaderTest.java

```diff
@@ -0,0 +1,146 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class TableColumnHeaderTest {
+
+    private MyTableColumnHeader tableColumnHeader;
+
+    @Before
+    public void beforeTest() {
+        tableColumnHeader = null;
+    }
+
+    @Test
+    public void test_resizeColumnToFitContent() {
+        ObservableList<Person> model = FXCollections.observableArrayList(
+                new Person("Humphrey McPhee", 76),
+                new Person("Justice Caldwell", 30),
+                new Person("Orrin Davies", 30),
+                new Person("Emma Wilson", 8)
+        );
+        TableColumn<Person, String> column = new TableColumn<>("Col ");
+        column.setCellValueFactory(new PropertyValueFactory<Person, String>("firstName"));
+        TableView<Person> tableView = new TableView<>(model) {
+            @Override
+            protected Skin<?> createDefaultSkin() {
+                return new TableViewSkin(this) {
+                    @Override
+                    protected TableHeaderRow createTableHeaderRow() {
+                        return new TableHeaderRow(this) {
+                            @Override
+                            protected NestedTableColumnHeader createRootHeader() {
+                                return new NestedTableColumnHeader(null) {
+                                    @Override
+                                    protected TableColumnHeader createTableColumnHeader(TableColumnBase col) {
+                                        tableColumnHeader = new MyTableColumnHeader(column);
+                                        return col == null || col.getColumns().isEmpty() || col == getTableColumn() ?
+                                                tableColumnHeader :
+                                                new NestedTableColumnHeader(col);
+
+                                    }
+                                };
+                            }
+                        };
+                    }
+                };
+            }
+        };
+
+        tableView.getColumns().add(column);
+
+        Toolkit tk = Toolkit.getToolkit();
+
+        Scene scene = new Scene(tableView);
+        Stage stage = new Stage();
+        stage.setScene(scene);
+        stage.setWidth(500);
+        stage.setHeight(400);
+        stage.centerOnScreen();
+        stage.show();
+
+        tk.firePulse();
+
+        double width = column.getWidth();
+        tableColumnHeader.resizeCol();
+        assertEquals("Width must be the same",
+                width, column.getWidth(), 0.001);
+
+        EventType<TableColumn.CellEditEvent<Person, String>> eventType = TableColumn.editCommitEvent();
+        column.getOnEditCommit().handle(new TableColumn.CellEditEvent<Person, String>(
+                tableView, new TablePosition<Person, String>(tableView, 0, column), (EventType) eventType, "This is a big text inside that column"));
```

Comment:
that looks like a rather convoluted way of changing content - what's wrong with straightforward changing the firstName of the first item :)

Reply:
Changing the firstName of the firstItem would be another test IMHO. It would test the fact that the backing model has changed and needs to be reflected in the view.

Here I want to test the fact that a user has modified a cell through the view, and it impacts the model.

Reply:
hmm ... tend to disagree: I see the resize feature/api as independent on how the data that is measured is changed. There's no automatic resize on editing (nor on changing content), you manually trigger the method, so keep the test as simple as possible.

Reply:
Allright, I can do it like that if you prefer

Reply:
ehh ... do what you want - I have no rights whatsoever to force (not even nudge :) you into anything. Just been there and thought you might want a helping hand.